File: back-end/src/main/java/com/huyphan/dtos/ChatMessageDto.java
package com.huyphan.dtos;

import com.huyphan.models.MessageContent;
import java.time.Instant;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ChatMessageDto {
    private long id;

    private MessageContentDto content;

    private Long conversationId;

    private String lastEditDate;
    private String sentDate;

    private boolean pinned;

    private boolean edited;

    private boolean deleted;
    private UserDto owner;
}

File: client/src/services/dtos/chat-message-dto.ts
import MessageContentDto from './message-content-dto';
import ReplyToMessageDto from './reply-to-message-dto';
import { UserDto } from './user-dto';

export default interface ChatMessageDto {
  readonly id: number;
  readonly content: MessageContentDto;
  readonly lastEditDate: string;
  readonly sentDate: string;
  readonly pinned: boolean;
  readonly edited: boolean;
  readonly owner: UserDto;
  readonly deleted: boolean;
  readonly conversationId: number;
  readonly replyToMessage?: ReplyToMessageDto;
}
